tests/into_iterator.rs

```diff
@@ -2,23 +2,64 @@
 #![allow(dead_code, unused_imports)]
 
 #[cfg(not(feature = "std"))]
+#[macro_use]
 extern crate alloc;
 
 #[cfg(not(feature = "std"))]
 use alloc::vec::Vec;
 
+use core::fmt::Debug;
+
 use derive_more::IntoIterator;
 
+fn test_into_iter<T: PartialEq + Debug, I: IntoIterator<Item = T>>(
+    iter: I,
+    vals: &[T],
+) {
+    assert_eq!(iter.into_iter().collect::<Vec<_>>(), vals);
```

Comment:
@MegaBluejay such unification is not good for tests, as doesn't allow us to understand well which test case is failing exactly: 
```
---- generic stdout ----
thread 'generic' panicked at 'assertion failed: `(left == right)`
  left: `[1, 2, 3]`,
 right: `[]`', tests/into_iterator.rs:18:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- generic_bounds stdout ----
thread 'generic_bounds' panicked at 'assertion failed: `(left == right)`
  left: `[1, 2, 3]`,
 right: `[]`', tests/into_iterator.rs:18:5

---- generic_refs stdout ----
thread 'generic_refs' panicked at 'assertion failed: `(left == right)`
  left: `[1, 2, 3]`,
 right: `[]`', tests/into_iterator.rs:18:5

---- generic_owned stdout ----
thread 'generic_owned' panicked at 'assertion failed: `(left == right)`
  left: `[1, 2, 3]`,
 right: `[]`', tests/into_iterator.rs:18:5

---- named_single stdout ----
thread 'named_single' panicked at 'assertion failed: `(left == right)`
  left: `[1, 2, 3]`,
 right: `[]`', tests/into_iterator.rs:18:5

---- tuple_single stdout ----
thread 'tuple_single' panicked at 'assertion failed: `(left == right)`
  left: `[1, 2, 3]`,
 right: `[]`', tests/into_iterator.rs:18:5
```

`#[track_caller]` may be helpful here to some degree, but not when multiple assertions are united in the same helper function.